src/Storage/Mapping/ContentTypeTitleTrait.php

```diff
@@ -69,8 +69,6 @@ protected function getTitleColumnNames($contentType)
         }
 
         // If this is a ContentType without any textfields
-        $keys = array_keys($fields);
-
-        return reset($keys);
+        return array_keys($contentType['fields']);
```

Comment:
Why is returning all fields then?

Reply:
The method itself returns an array of fieldnames that are considered to cobble together the title. If _no_ suitable column is found at all, it'll either display `(no title)` or something put together from fields. For example: 

```
dummy:
    name: Dummies
    singular_name: Dummy
    fields:
        turtlesallthewaydown:
            type: select
            values: [ A-tuin, Donatello, Rafael, Leonardo, Michelangelo, Koopa, Squirtle ]
        image:
            type: image
        datum:
            type: date
```

No `title` field, no `title_format` and no `type: text` fields. This CT would show "Donatello" or "Koopa" or the image filename in the listing overview, and "latest changes". It's not ideal, but IMHO better than before. 

Before: 

![screen shot 2017-10-20 at 14 10 29](https://user-images.githubusercontent.com/1833361/31820248-ca586f42-b5a0-11e7-82a4-fa011f441a2f.png)


After: 

![screen shot 2017-10-20 at 14 12 31](https://user-images.githubusercontent.com/1833361/31820252-ce4fb416-b5a0-11e7-9515-2b05c627e830.png)

Reply:
I can't begin to tell you how wrong I find this approach :cry:

Reply:
Also, the change introduced in #7067 tries to return a non-existing variable … This needs tests!

Reply:
> Also, the change introduced in #7067 tries to return a non-existing variable

Yes, that is exactly what this PR is fixing: that piece of code that came from elsewehere wasn't adapted properly. 

> This needs tests!

When we discussed this, I thought we decided that we'd just fix it up to fix the bugs we're seeing, since it is marked "deprecated" and will get replaced sooner or later regardless.